controlpanel/api/models/user.py

```diff
@@ -83,10 +83,28 @@ def reset_mfa(self):
         auth0.ManagementAPI().reset_mfa(self.auth0_id)
 
     def save(self, *args, **kwargs):
+        request_username = None
+        request = CrequestMiddleware.get_request()
+        if request:
+            request_username = request.user.username
+
         try:
-            User.objects.get(pk=self.pk)
+            existing = User.objects.get(pk=self.pk)
+
         except User.DoesNotExist:
             cluster.User(self).create()
+            if self.is_superuser:
+                slack.notify_team(
+                    slack.CREATE_SUPERUSER_MESSAGE.format(username=self.username),
+                    request_username=request_username,
+                )
+
+        else:
+            if self.is_superuser and not existing.is_superuser:
+                slack.notify_team(
```

Comment:
If we go for having this like this I think would be more clear to have the use of the `existing` variable closer to the query rather than in this `else` block, e.g.:

```python
try:
    existing = User.objects.get(pk=self.pk)
    if self.is_superuser and not existing.is_superuser:
        # stuff in `else` block
except User.DoesNotExist:
    # ... same
```

Reply:
No, putting more code into the `try...except` block is worse, because we are not handling any exceptions raised in that code.

Reply:
Yeah but the if/sent notification in the `try` is the happy path, that's not exception handling, that's normal flow.

To be honest I personally don't like overusing exceptions for normal control flow like this, this could be written without having to use `try`/`catch` and it could be more readable, e.g.:

```python
if existing = User.objects.filter(pk=self.pk).first():
  # send notification about existing user becoming superuser
else:
  # send notification about new superuser user created
```

IMHO of course.

Reply:
Using exceptions for flow control is Pythonic.

Reply:
But I agree, your way is more readable.